### PR TITLE
airbyte-ci: add force release flag to breaking changes

### PR DESCRIPTION
**Pull request by lazebnyi**
## What
Sometimes we need to release breaking changes faster than once a week, but with the current CI check, it is possible only with force merge.

## How
Add `forceRelease` flag to skip `upgradeDeadline` date validation

## User Impact
No

## Can this PR be safely reverted and rolled back?
- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorReleases.yaml [hunk 1]
```diff
@@ -38,6 +38,9 @@ definitions:
       message:
         description: Descriptive message detailing the breaking change.
         type: string
+      forceRelease:
+        description: The flag used to indicate whether a breaking change in the connector version should take effect less tha one week, bypassing the standard deadline requirement.
```
Comment (natikgadzhi) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorReleases.yaml:42]:
```suggestion
        description: The flag used to indicate whether a breaking change in the connector version should take effect less than one week, bypassing the standard deadline requirement.
```

**Review by natikgadzhi — COMMENTED** [review]
I'm a bit concerned that we're adding a field for `forceRelease` that in this PR only affects the connectors_qa CI check.

It does not actually make the platform to force release this connector to all affected workspaces (which is probably good). 

Wouldn't a better approach be a way to ignore the check specifically in connectors_qa invocation, if it's required? 

I.e. maybe we can [skip-breaking-change-deadline] in the commit message, or anywhere in PR body, and pass that to the check?

File: docs/connector-development/connector-metadata-file.md [hunk 2]
```diff
@@ -129,6 +133,20 @@ releases:
       upgradeDeadline: "2023-12-31"
 ```
 
+#### `forceRelease`
+The optional `scopedImpact` property allows you to bypass check that the `upgradeDeadline` is set to at least one week in the future.
```
Comment (clnoll) [inline docs/connector-development/connector-metadata-file.md:137]:
```suggestion
The optional `forceRelease` property allows you to bypass check that the `upgradeDeadline` is set to at least one week in the future.
```

File: docs/connector-development/connector-metadata-file.md [hunk 3]
```diff
@@ -129,6 +133,20 @@ releases:
...
+
+Here's an example:
```
Comment (clnoll) [inline docs/connector-development/connector-metadata-file.md:139]: Can you add some examples for which `forceRelease` is appropriate versus inappropriate?

**Review by clnoll — COMMENTED** [review]
A couple of minor requests from me @lazebnyi.

Thinking about some of @natikgadzhi 's comments - 

> I'm a bit concerned that we're adding a field for forceRelease that in this PR only affects the connectors_qa CI check.
> It does not actually make the platform to force release this connector to all affected workspaces (which is probably good).
> Wouldn't a better approach be a way to ignore the check specifically in connectors_qa invocation, if it's required?
> I.e. maybe we can [skip-breaking-change-deadline] in the commit message, or anywhere in PR body, and pass that to the check?

Having the `forceRelease` in the file gives us the benefit of having the history that the release was forced and why. If we override the check, this information is a little more difficult to find. 

I think that to understand the best approach I'd want to know what sort of problems we are solving with this. Is it primarily that we need to be able to release breaking changes to fix something that we broke? Or is it more about the fact that there may be some minor changes, or changes that affect few users and we don't see a reason to wait?

**Comment by natikgadzhi** [comment]
Good point re: history.

**Pull request closed without merging**